tools/ci/check_grep_secret.sh

```diff
@@ -38,4 +38,20 @@ if grep -P 'playsound\(([^,]*), "(sound\/[^\[]+)"' code/**/*.dm;	then
     st=1
 fi;
 
+if grep -P 'plane\s*=\s*[0-9]+|plane\s*=\s*[A-Z_]+\s*[+\-*]\s*' */**/*.dm;	then
+    echo "ERROR: don't directly set plane to a number, please use a define."
+    st=1
+fi;
+
+if grep -P 'rand\([^)]*[0-9]\.' */**/*.dm;	then
+    echo "ERROR: rand() does not support floating point numbers, use randfloat() instead."
+    st=1
+fi;
+
+
+if grep -P '^ABSTRACT_TYPE\([^/]' */**/*.dm;	then
+    echo "ERROR: You need to include the slash before the area type name in ABSTRACT_TYPE."
+    st=1
+fi;
+
```

Comment:
🤓 erm, 🤓 please atomize this PR 🤓 these are changes unrelated to the title 🤓 btw

Reply:
:godmode: